WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -566,30 +567,84 @@ public static void viewHistoryDetailForResult(Activity activity, Revision revisi
      * Load the post preview as an authenticated URL so stats aren't bumped
      */
     public static void browsePostOrPage(Context context, SiteModel site, PostModel post) {
-        if (site == null || post == null || TextUtils.isEmpty(post.getLink())) {
+        browsePostOrPageEx(context, site, post, RemotePreviewType.NOT_A_REMOTE_PREVIEW);
+    }
+
+    public static void previewPostOrPageForResult(
+            Activity activity,
+            SiteModel site,
+            PostModel post,
+            RemotePreviewType remotePreviewType
+    ) {
+        browsePostOrPageEx(activity, site, post, remotePreviewType);
+    }
+
+    private static void browsePostOrPageEx(
+            Context context,
+            SiteModel site,
+            PostModel post,
+            RemotePreviewType remotePreviewType) {
+        if (
```

Comment:
Nitpicking: do you mind splitting this in 2 tests for readability / style.

Reply:
👍 done here 2ff1e77e6da180b8e89f08d725ea9a18117f5f39